src/algorithms/tracking/TrackSeeding.cc

```diff
@@ -86,8 +86,10 @@ std::vector<edm4eic::TrackParameters*> eicrecon::TrackSeeding::makeTrackParams(S
       float R = std::get<0>(RX0Y0);
       float X0 = std::get<1>(RX0Y0);
       float Y0 = std::get<2>(RX0Y0);
-      if (R > std::sqrt(std::cbrt(std::numeric_limits<float>::max()))) {
-        // avoid future float overflow for hits on a line
+      if (R > std::numeric_limits<float>::max() ||
+	std::abs(X0) > std::numeric_limits<float>::max() ||
+	std::abs(Y0) > std::numeric_limits<float>::max()) {
+        // avoid float overflow for hits on a line
```

Comment:
I think the pathological cases (straight line, circle around origin) need to be caught before entering this function, not in the function. It will be easier to handle them in the caller than here. If you have a straight line, the perigee is different but still determined. if you have a center point at the origin, you don't need to assign an invalid point but discard the seed.

Reply:
I renamed `findRoot `to `findPCA` (for point-of-closest approach). I also moved the check on the circle center point to before the calling of this function.

Following the discussion at the track reconstruction meeting on Thursday, we can add additional functionality to handle the linear case later. We will also consider a modification to using Acts functions to find the seed parameters in the future.